sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -656,8 +657,11 @@ public void deleteDataset(String projectId, String datasetId)
     }
 
     @VisibleForTesting
-    long insertAll(TableReference ref, List<TableRow> rowList, @Nullable List<String> insertIdList,
-        BackOff backoff, final Sleeper sleeper) throws IOException, InterruptedException {
+    long insertAll(TableReference ref, List<ValueInSingleWindow<TableRow>> rowList,
```

Comment:
I actually tried the list of indices first, but abandoned it because the code complexity grew too much (it's quite difficult for insertAll to keep track of the failed index today, and I'd prefer not to rewrite that function completely - there have been multiple subtle bugs in the past). Also I think this is a better abstraction - WindowedValues are used everywhere else, and we may well use them in the future in insertAll (e.g. for partitioned tables).